src/test/java/ru/ayaz/UserMessageHandlerTest.java

```diff
@@ -0,0 +1,127 @@
+package ru.ayaz;
+
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class UserMessageHandlerTest {
+
+    private UserMessageHandler messageHandler;
+
+    @Before
+    public final void before() {
+        int queueSize = 500;
+        messageHandler = new UserMessageHandler(queueSize);
+    }
+
+    @Test
+    public void shouldPutMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+
+        messageHandler.enqueueMessage(someMessage);
+
+        assertTrue(messageHandler.queueContainsMessage(someMessage));
+    }
+
+    @Test
+    public void shouldTakeMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+        messageHandler.enqueueMessage(someMessage);
+
+        UserMessage gottenMessage = messageHandler.takeMessage();
+
+        assertEquals(someMessage, gottenMessage);
+    }
+
+
+    @Test
+    public void shouldInvokePrintUnknownCommandWhenMaliformedCommandPassed() throws InvalidNicknameException, InterruptedException {
+        User user = new User();
+        user.setNickname("Ayaz");
+        UserMessage commandMessage = new UserMessage("Ayaz", "#quiit");
+        PrintWriter writer = mock(PrintWriter.class);
+        UserMessageHandler spyMessageHandler = spy(messageHandler);
+        spyMessageHandler.addUserToMap(user);
+        spyMessageHandler.addWriterToMap(user.getNickname(), writer);
+        doReturn(commandMessage).doThrow(InterruptedException.class).when(spyMessageHandler).takeMessage();
```

Comment:
No, what I mean is "what this line of code is meant to be doing ?". I didn't understand whether it should return or throw when "takeMessage" method is called.
From you comment I take it that this line says: "On the first call return, on the second call throw", is that so ?